### PR TITLE
SearchBar refactoring

### PR DESCRIPTION
**Pull request by salesfelipe**
#### What is the purpose of this pull request?
Refact the SearchBar

#### What problem is this solving?
Adapting component to follow the  flux architecture pattern

#### How should this be manually tested?
[workspace](https://felipe--storecomponents.myvtex.com)

#### Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Changed
+- `SearchBar` refactoring
```
Comment (brunojdo) [inline CHANGELOG.md]: Refact `SearchBar` component.

File: react/components/SearchBar/components/ResultsList.js [hunk 2]
```diff
@@ -65,25 +69,39 @@ class ResultsList extends Component {
     return (
       <div className={listClassNames}>
         <Link
-          onClick={() => { this.props.closeMenu() }}
+          onClick={() => {
```
Comment (brunojdo) [inline react/components/SearchBar/components/ResultsList.js]: adjust your lint

File: CHANGELOG.md (hunk 1 above)
Comment (salesfelipe) [reply CHANGELOG.md]: 👍

File: react/components/SearchBar/components/ResultsList.js (hunk 2 above)
Comment (salesfelipe) [reply react/components/SearchBar/components/ResultsList.js]: 👍

**Pull request merged**